src/System.Drawing.Common/src/System/Drawing/Printing/PreviewPrintController.Windows.cs

```diff
@@ -15,63 +13,38 @@ namespace System.Drawing.Printing
     /// <summary>
     /// A PrintController which "prints" to a series of images.
     /// </summary>
-    public class PreviewPrintController : PrintController
+    public partial class PreviewPrintController : PrintController
     {
-        private IList _list = new ArrayList(); // list of PreviewPageInfo
-        private System.Drawing.Graphics _graphics;
+        private Graphics _graphics;
         private DeviceContext _dc;
-        private bool _antiAlias;
-
-        private void CheckSecurity()
-        {
-        }
-
-        /// <summary>
-        /// This is new public property which notifies if this controller is used for PrintPreview.
-        /// </summary>
-        public override bool IsPreview
-        {
-            get
-            {
-                return true;
-            }
-        }
 
         /// <summary>
         /// Implements StartPrint for generating print preview information.
         /// </summary>
         public override void OnStartPrint(PrintDocument document, PrintEventArgs e)
         {
-            Debug.Assert(_dc == null && _graphics == null, "PrintController methods called in the wrong order?");
-
-            // For security purposes, don't assume our public methods are called in any particular order
-            CheckSecurity();
-
             base.OnStartPrint(document, e);
 
             if (!document.PrinterSettings.IsValid)
+            {
                 throw new InvalidPrinterException(document.PrinterSettings);
+            }
 
             // We need a DC as a reference; we don't actually draw on it.
             // We make sure to reuse the same one to improve performance.
-            _dc = document.PrinterSettings.CreateInformationContext(modeHandle);
+            _dc = document.PrinterSettings.CreateInformationContext(_modeHandle);
         }
 
         /// <summary>
         /// Implements StartEnd for generating print preview information.
         /// </summary>
         public override Graphics OnStartPage(PrintDocument document, PrintPageEventArgs e)
         {
-            Debug.Assert(_dc != null && _graphics == null, "PrintController methods called in the wrong order?");
```

Comment:
Why remove this assert?

Reply:
Easily triggered by calling without before calling `OnStartPrint`

Reply:
Should we instead throw an InvalidOperationException?

Reply:
We could - it would be breaking and require us to add a new exception message. What do you think?

Reply:
@stephentoub what do you think?

Reply:
What's the current release behavior in a situation where this assert would have been triggered?

Reply:
From taking a look at the code I can see that we would Null Ref here: https://github.com/dotnet/corefx/pull/36736/files/558de14d62537ace81993796d54954b9c4b613f5#diff-368f90b5da0f52ee19e0acf2a4a54780R76 when calling: `_dc.Hdc`